test/success/derived-rule.prl

```diff
@@ -17,42 +17,42 @@ Rule MyRule : [
 // it replaces the current goal with the premises induced by the derived rule. The
 // hypothetical contexts are also unified together in the appropriate way (i.e. we 
 // do not require strict equality of contexts, but only compatibility).
-Thm MyCorollary : [ bool -> bool ] by [
+Thm MyCorollary : [ (-> bool bool) ] by [
```

Comment:
@favonia Because in this PR I have switched RedPRL's term language to use an s-expression style notation. The reason for this is that sexprs are possible to layout in a civilized way when pretty-printing, whereas the notation we used before was basically hopeless in this respect.